hoodie-client/src/main/java/com/uber/hoodie/func/payload/HoodieRecordBufferedIteratorPayload.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.func.payload;
+
+import com.uber.hoodie.common.model.HoodieRecord;
+import com.uber.hoodie.common.model.HoodieRecordPayload;
+import java.util.Optional;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.IndexedRecord;
+
+public class HoodieRecordBufferedIteratorPayload<T extends HoodieRecordPayload>
+    extends AbstractBufferedIteratorPayload<HoodieRecord<T>, Optional<IndexedRecord>> {
+
+  // It caches the exception seen while fetching insert value.
+  public Optional<Exception> exception = Optional.empty();
+
+  public HoodieRecordBufferedIteratorPayload(HoodieRecord record, Schema schema) {
+    this.inputPayload = record;
+    try {
+      this.outputPayload = record.getData().getInsertValue(schema);
```

Comment:
An observation:  I am assuming this serialization is cpu intensive and with your changes, this work is decoupled from parquet write for new file-group generation case (Hoodie Create Handle). For the merge-case though, the deserialization seems to be tied up with parquet data fetch  but they together are decoupled from parquet write.

Reply:
Yes, that's true. The issue is that for CreateHandle the process isn't dependent on anything apart from the input, whereas in the MergeHandle, a merge process with the new records takes place. A larger refactor could help address this but I'm hoping to keep that for later.

Reply:
lets capture the AIs from here into an issue?

Reply:
https://github.com/uber/hudi/issues/374